packages/config/test/ast-utils/addToCypressConfig.spec.ts

```diff
@@ -19,223 +19,374 @@ const { addTestingTypeToCypressConfig } = proxyquire('../../src/ast-utils/addToC
+              bundler: 'webpack',
+              framework: 'react',
+              bundlerConfigPath: './webpack.config.js',
+              needsExplicitConfig: true,
+            },
+            isProjectUsingESModules: false,
+          })
+
+          expect(stub.getCall(0).lastArg.trim()).to.eq(dedent`
+          const { defineConfig } = require("cypress");
+
+          const webpackConfig = require("./webpack.config.js");
+
+          module.exports = defineConfig({
+            component: {
+              devServer: {
+                framework: "react",
+                bundler: "webpack",
+                webpackConfig,
+              },
+            },
+          });`.trim())
+
+          expect(result.result).to.eq('ADDED')
+        })
+      })
+
+      context('without bundleConfigPath', () => {
+        it('adds component testing to e2e config', async () => {
+          const result = await addTestingTypeToCypressConfig({
+            filePath: path.join(__dirname, '../__fixtures__/has-e2e.config.js'),
+            info: {
+              testingType: 'component',
+              bundler: 'webpack',
+              framework: 'react',
+              needsExplicitConfig: true,
+              bundlerConfigPath: undefined,
+            },
+            isProjectUsingESModules: true,
+          })
+
+          // we generate this using Babel which intentionally does **NOT**
+          // handle formatting, so the output format is kind of weird.
+          // we rely on the user's eslint or prettier to format this properly.
+          const expected = dedent`
+          module.exports = {
+            e2e: {},
+          
+            component: {
+              devServer: {
+                framework: "react",
+                bundler: "webpack",
+                // provide your webpack config here...
+                // webpackConfig,
+              },
+            },
+          };`
+
+          expect(stub.getCall(0).lastArg.trim()).to.eq(expected)
+
+          expect(result.result).to.eq('MERGED')
+        })
+      })
+
+      context('with bundleConfigPath', () => {
+        it('adds component testing to e2e config', async () => {
+          const result = await addTestingTypeToCypressConfig({
+            filePath: path.join(__dirname, '../__fixtures__/has-e2e.config.js'),
+            info: {
+              testingType: 'component',
+              bundler: 'webpack',
+              framework: 'react',
+              needsExplicitConfig: true,
+              bundlerConfigPath: './webpack.config.js',
+            },
+            isProjectUsingESModules: false,
+          })
+
+          // we generate this using Babel which intentionally does **NOT**
+          // handle formatting, so the output format is kind of weird.
+          // we rely on the user's eslint or prettier to format this properly.
+          const expected = dedent`
+          const webpackConfig = require('./webpack.config.js')module.exports = {
```

Comment:
This seems like unexpected behavior and I feel like we should be able to handle this better - if we're adding the import, could we just consistently add it to the top of the config file with a newline or something?

Reply:
Babel does not support adding formatting what-so-ever, unfortunately. We should be able to run it through something like `@babel/generator`, I'll look into this.

Reply:
Oh, easy... recast has a `prettyPrint`, let's just use that. It also solves the "what if they don't have prettier" problem - the basic `prettyPrint` is quite good.